src/RestaurantPage.jsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router';
+
+import {
+  loadRestaurant,
+} from './actions';
+
+import { get } from './utils';
+
+export default function RestaurantPage() {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    const restaurant = useSelector(get('restaurant'));
+    const { restaurantId } = useParams();
+    dispatch(loadRestaurant(restaurantId));
+  });
```

Comment:
```suggestion
  const restaurant = useSelector(get('restaurant'));
  
  const { restaurantId } = useParams();
  
  useEffect(() => {  
    dispatch(loadRestaurant(restaurantId));
  }, []);
```

useEffect를 사용하실 때, 빈 배열을 추가해서 한 번만 호출하도록 해야할 것 같아요. 그리고 useSelector와 useParams는 useEffect안에 없어야 될 것 같습니다.